### PR TITLE
Map updates

### LINKED ISSUE #46: Add TIP project linear features
Include TIP Projects Linear Features as a map layer along with markers. 
Several TIP projects extent for several miles and can be miss leading have a simple marker in a location that does not capture the full extent of the project  

If not possible with current DRAFT version of web mapping application, one solution could be to include a link for users to explore the GIS data sets in a simple AGO web map. 
screen shot 2018-05-02 at 11 12 00 am

### PR DESCRIPTION
**Pull request by rbeatty1**
Close #46 and #81

### CODE REVIEW

File: src/components/map/Map.js [hunk 1]
```diff
@@ -456,12 +460,19 @@ class MapComponent extends Component {
...
-        "all",
-        this.state.catFilter,
-        this.state.keyFilter
-      ]);
+    // the filter wasn't working because this.map.setFilter was being called immediately after setting category state,
+    // which is async, and so there was a disconnect between what category state was and what map.setFilter was pulling from
+    // moving setFilter to the render method ensures it will always be filtering the correct state
+    if (this.map) {
+      ["pa-tip-points", "pa-tip-lines"].forEach(layer => {
+        if (this.map && this.state.catFilter && this.state.keyFilter) {
```
Comment (DVRPCfreight) [inline src/components/map/Map.js]: Can we switch line 467 and 468? Do not run the loop until we are sure the map and states are defined.

File: src/utils/tileGeometryColorType.js [hunk 2]
```diff
@@ -58,8 +58,9 @@ export const colors = {
 };
 
 export const geometryColorType = project => {
+  console.log({ project });
```
Comment (DVRPCfreight) [inline src/utils/tileGeometryColorType.js]: Clean up

File: src/components/map/Map.js [hunk 3]
```diff
@@ -456,12 +460,19 @@ class MapComponent extends Component {
...
   render() {
-    if (this.map && this.state.catFilter && this.state.keyFilter) {
-      this.map.setFilter("pa-tip-projects", [
-        "all",
-        this.state.catFilter,
-        this.state.keyFilter
-      ]);
+    // the filter wasn't working because this.map.setFilter was being called immediately after setting category state,
+    // which is async, and so there was a disconnect between what category state was and what map.setFilter was pulling from
+    // moving setFilter to the render method ensures it will always be filtering the correct state
```
Comment (mmolta) [inline src/components/map/Map.js]: remove comments

File: src/components/map/Map.js (hunk 1 above)
Comment (mmolta) [inline src/components/map/Map.js]: make sure everything is fully rendered before filtering. Could do:
`if(this.map && this.state.catFilter && this.state.keyFilter) {
    ["pa-tip-points", "pa-tip-lines"].forEach(layer => {
        ...
    })
}
`

File: src/components/tiles-container/TilesContainer.js [hunk 4]
```diff
@@ -131,14 +128,12 @@ class TilesContainer extends Component {
             projects.map(feature => (
               <ListItem
                 data={feature}
-                key={feature.attributes.OBJECTID}
+                key={feature.mpms}
```
Comment (mmolta) [inline src/components/tiles-container/TilesContainer.js]: if OBJECTID is still available, switch back to it. Keys have to be unique

File: src/components/tiles-container/TilesContainer.js [hunk 5]
```diff
@@ -131,14 +128,12 @@ class TilesContainer extends Component {
...
                 length={projects.length}
               />
             ))
           ) : (
-            projects.map(feature => (
-              <Tile data={feature} key={feature.attributes.OBJECTID} />
-            ))
+            projects.map(feature => <Tile data={feature} key={feature.mpms} />)
```
Comment (mmolta) [inline src/components/tiles-container/TilesContainer.js]: same as line 134 - make sure keys are unique

File: src/components/tiles-container/TilesContainer.js (hunk 4 above)
Comment (rbeatty1) [reply src/components/tiles-container/TilesContainer.js]: Will check within mapbox gl feature for unique id, no need to have any AGO jawns weaseling back in.

**Pull request merged**